kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/targetDefaultForClassName.kt

```diff
@@ -48,6 +49,10 @@ actual inline fun <reified A> targetDefaultForClass(): Arb<A>? {
       A::class == LocalTime::class -> Arb.localTime() as Arb<A>
       A::class == Period::class -> Arb.period() as Arb<A>
       A::class == BigDecimal::class -> Arb.bigDecimal() as Arb<A>
+      A::class.isData -> {
```

Comment:
@sksamuel I wonder would it be useful to provide users the ability to register / inject their own data class arbs?

Reply:
Easy enough to do that by just specifying the arb to the checkAll method like `checkAll(1000, Arb.myDataArb())`
This is for when you want to do `checkAll<String, Foo> { ... }`
But with your recent idea who knows how checkAll will advance.